src/param_utils.cpp

```diff
@@ -43,5 +46,31 @@ void sort_params(std::vector<instruction_ref>& params)
               }));
 }
 
+std::vector<instruction_ref>
+find_inputs(const std::unordered_map<instruction_ref, instruction_ref>& map_ins,
```

Comment:
This function needs a descriptive comment or no one else will ever be able to use it.